GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            get { return BundleBrowserIsPresent(); }
+        }
+
+        /// <summary>
+        /// The detected version of Asset Bundle Browser.
+        /// </summary>
+        public static string AssetBundleBrowserVersion
+        {
+            get { return GetBrowserVersion(); }
+        }
+
+
+        // Detects AssetBundleBrowser Namespace and the AssetBundleBrowserMain Class
+        private static bool BundleBrowserIsPresent(bool useCurrentValueIfPresent = true)
+        {
+            if (useCurrentValueIfPresent && _assetBundleBrowserIsPresent.HasValue)
+            {
+                return _assetBundleBrowserIsPresent.Value;
+            }
+
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                // Checking that assembly != null since we do not want to rely on iteration through
+                // AppDomain.CurrentDomain.GetAssemblies() to yield non-null items for this function to work
+                if (assembly == null)
+                {
+                    continue;
+                }
+
+                foreach (var type in assembly.GetTypes())
+                {
+                    // Look for AssetBundleBrowserMain in the AssetBundleBrowser Namespace
+                    if (type != null && string.Equals(type.Namespace, "AssetBundleBrowser") &&
+                        type.Name.Equals("AssetBundleBrowserMain"))
+                    {
+                        _assetBundleBrowserIsPresent = true;
+                        return _assetBundleBrowserIsPresent.Value;
+                    }
+                }
+            }
+
+            _assetBundleBrowserIsPresent = false;
+            return false;
+        }
+
+        // Evaluates whether a folder has "AssetBundles-Browser" in its name, making it
+        // a candidate for the Asset Bundles Browser Folder
+        private static bool IsAssetBundleBrowserFolder(string folderName)
+        {
+            var regex = new Regex(@"AssetBundles-Browser", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+            return regex.Matches(folderName).Count > 0;
+        }
+
+        // Extracts AssetBundleBrowser version name from the Asset Bundle Browser package.json file
+        private static string GetBrowserVersion(bool useCurrentValueIfPresent = true)
+        {
+            if (useCurrentValueIfPresent && _assetBundleBrowserVersion != null)
+            {
+                return _assetBundleBrowserVersion;
+            }
+
+            var assetsPath = Application.dataPath;
+            var assetBundleBrowserFolderPaths =
+                Directory.GetDirectories(assetsPath).ToArray().Where(IsAssetBundleBrowserFolder);
+
+            foreach (var folderPath in assetBundleBrowserFolderPaths)
+            {
+                var expectedPackageDotJsonPath = Path.Combine(folderPath, "package.json");
+                if (!File.Exists(expectedPackageDotJsonPath))
+                {
```

Comment:
Log warning?

Reply:
Resolved.
Added a Debug.LogWarning just so the the developer knows we attempted to find the file